src/producer.rs

```diff
@@ -254,20 +257,19 @@ impl<P: Partitioner> Producer<P> {
               V: AsBytes
     {
         let mut rs = try!(self.send_all(ref_slice(rec)));
-        if self.config.required_acks == 0 {
+
+        if self.config.required_acks != 0 {
             // ~ with no required_acks we get no response and
             // consider the send-data request blindly as successful
-            Ok(())
-        } else {
             assert_eq!(1, rs.len());
-            rs.pop().unwrap().offset.map(|_| ())
+            rs.pop().unwrap();
         }
+
+        Ok(())
```

Comment:
Hmm, I did just have a thought, though. It seems `Producer::send_all` returns the response from the broker, while `Producer::send` just returns `()`. What if the user would want to access the information in the returned `ProduceConfirm`, such as to know the offset of the new message? Do you think it would make sense to return a `Result<ProduceConfirm>` instead?